src/Platform/Microsoft.Testing.Platform/OutputDevice/Console/TestWorkerFrame.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Globalization;
+
+namespace Microsoft.Testing.Platform.OutputDevice.Console;
+
+/// <summary>
+/// Captures <see cref="TestWorker"/> that was rendered to screen, so we can only partially update the screen on next update.
+/// </summary>
+internal sealed class TestWorkerFrame
+{
+    private const int MaxColumn = 120;
+
+    private readonly (TestWorker TestWorkerProgress, int DurationLength)[] _workers;
+
+    public int Width { get; }
+
+    public int Height { get; }
+
+    public int ProgressCount { get; private set; }
+
+    public TestWorkerFrame(TestWorker?[] nodes, int width, int height)
+    {
+        Width = Math.Min(width, MaxColumn);
+        Height = height;
+
+        _workers = new (TestWorker, int)[nodes.Length];
+
+        foreach (TestWorker? status in nodes)
+        {
+            if (status is not null)
+            {
+                _workers[ProgressCount++].TestWorkerProgress = status;
+            }
+        }
+    }
+
+    public void AppendTestWorkerProgress(int i, AnsiTerminal terminal)
+    {
+        TestWorker p = _workers[i].TestWorkerProgress;
+
+        string durationString = $" ({p.Stopwatch.Elapsed.TotalSeconds:F1}s)";
+
+        _workers[i].DurationLength = durationString.Length;
+
+        int passed = p.Passed;
+        int failed = p.Failed;
+        int skipped = p.Skipped;
+
+        string? detail = !RoslynString.IsNullOrWhiteSpace(p.Detail) ? $"- {p.Detail}" : null;
+        terminal.Append('[');
+        terminal.SetColor(TerminalColor.DarkGreen);
+        terminal.Append("✅");
+        terminal.Append(passed.ToString(CultureInfo.CurrentCulture));
+        terminal.ResetColor();
+
+        terminal.Append("/");
+
+        terminal.SetColor(TerminalColor.DarkRed);
+        terminal.Append("❌");
+        terminal.Append(failed.ToString(CultureInfo.CurrentCulture));
+        terminal.ResetColor();
+
+        terminal.Append("/");
+
+        terminal.SetColor(TerminalColor.DarkYellow);
+        terminal.Append("❔");
+        terminal.Append(skipped.ToString(CultureInfo.CurrentCulture));
+        terminal.ResetColor();
+        terminal.Append(']');
+
+        terminal.Append(' ');
+        terminal.Append(p.AssemblyName);
+        terminal.Append('(');
+        terminal.Append(p.TargetFramework);
+        terminal.Append('|');
+        terminal.Append(p.Architecture);
+        terminal.Append(')');
+        if (!RoslynString.IsNullOrWhiteSpace(detail))
+        {
+            terminal.Append(" - ");
+            terminal.Append(detail);
+        }
+
+        terminal.SetCursorHorizontal(Width - durationString.Length);
+        terminal.Append(durationString);
```

Comment:
Maybe a clock icon for duration could be good too.

Reply:
If you want to go back to emojis fully I can add it. But to me the emojis steal too much focus in the output.